test/lint/lint-circular-dependencies.sh

```diff
@@ -14,9 +14,12 @@ EXPECTED_CIRCULAR_DEPENDENCIES=(
     "blockchain/blockchain_genesis -> blockchain/blockchain_parameters -> blockchain/blockchain_genesis"
     "blockchain/blockchain_parameters -> settings -> key_io -> chainparams -> blockchain/blockchain_parameters"
     "blockdb -> validation -> injector -> blockdb"
+    "blockencodings -> injector -> p2p/graphene_receiver -> net_processing -> blockencodings"
+    "blockencodings -> injector -> p2p/graphene_sender -> blockencodings"
     "chainparamsbase -> util -> chainparamsbase"
     "coins -> snapshot/iterator -> coins"
     "coins -> snapshot/messages -> coins"
+    "core_io -> script/sign -> policy/policy -> validation -> core_io"
```

Comment:
And in reference to the comment, which I guess is more an observation: Here we did introduce an ordinary, non-injector-induced circular dependency.